main/modes/games/cGrove/Garden/cg_Grove.c

```diff
@@ -881,14 +881,13 @@ static void cg_handleInputGarden(cGrove_t* cg)
         {
             if (getTouchJoystick(&phi, &r, &intensity))
             {
-                int16_t speed = phi >> 5;
-                if (!(speed <= 5))
+                if (r >= 500)
                 {
                     /* ESP_LOGI("CG", "touch center: %" PRIu32 ", intensity: %" PRIu32 ", intensity %" PRIu32, phi, r,
-                     * intensity); */
+                     intensity);  */
                     // Move hand
-                    cg->grove.cursor.pos.x += (getCos1024(phi) * speed) / 1024;
-                    cg->grove.cursor.pos.y -= (getSin1024(phi) * speed) / 1024;
+                    cg->grove.cursor.pos.x += (getCos1024(phi) * (r >> 5)) / 1024;
```

Comment:
Maybe a bit late, but there is a function to do this for you, [getTouchCartesian()](https://adam.feinste.in/Super-2024-Swadge-FW/touchUtils_8c.html#a3d0b9236fd51fbfa13444d80048b1d02).

Reply:
Nice. I saw the function, decided I didn't want to figure it out, and then screwed up.

Oh well.

Reply:
It's a learning experience.